src/components/matching/PreferenceDragAndDrop.tsx

```diff
@@ -0,0 +1,89 @@
+import { useCallback } from "react";
+import {
+  DragDropContext,
+  Draggable,
+  Droppable,
+  type DropResult,
+} from "react-beautiful-dnd";
+import DragHandleIcon from "@mui/icons-material/DragHandle";
+import { type Preference } from "./MatchingFormSteps";
+
+const PreferenceDragAndDrop = ({
+  preferences,
+  setPreferences,
+}: {
+  preferences: Preference[];
+  setPreferences: React.Dispatch<React.SetStateAction<Preference[]>>;
+}) => {
```

Comment:
Instead of setPreferences, can we do `onChange: (Preference[]) => void` to make it more consistent with other input components